docs/DesignDocs/AttestationEvidenceFormatSupport.md

```diff
@@ -0,0 +1,118 @@
+Attestation Evidence Format Support
+=====
+
+One of the goal of OE SDK attestation design is to be TEE-agnostic. At the same time, OE SDK should enable a verifier application to verify evidences from different sources and TEEs, if the format is known to the verifier.
+
+Motivation
+----------
+
+Current evidence verify API `oe_verify_evidence()` verifies the evidences generated by `oe_get_evidence()` and returns the claims contained in the evidence. Such evidence has a specific format including an `oe_attestation_header`.
+As reported in issue #2999, attestation service has to verify evidences from different sources other than those generated by `oe_get_evidence()`. Such as evidence generated by legacy OE SDK API `oe_get_report()` and raw TEE evidence generation such as SGX quote generation mechanism. And like `oe_verify_evidence()` returns claims, attestation service is also required to return the supplemental information. To achieve that, a new verify API is needed to verify different evidence structures and return the required supplemental information.
+
+Goals:
+- Verifier can call OE SDK API to verify known evidence formats generated by either OE SDK API or other attesters.
+
+User Experience
+---------------
+
+New parameter `format_id` will be added to `oe_verify_evidence` to verify various evidences other than the ones generated by `oe_get_evidence()`, such as a raw SGX quote. To verify an evidence generated by `oe_get_evidence()`, `format_id` should be `null`.
+
+Scenario 1 - Verifier verifies a raw evidence (not generated by `oe_get_evidence()`) of known format id:
+
+1. Call OE SDK public API to initialize verifier plugins.
+2. Get the raw evidence and its format id from an attester, such as get a raw sgx quote from Intel SGX SDK.
+3. Call the new API with the evidence and the format id to verify the evidence.
+
+Scenario 2 - Verifier verifies an evidence generated by `oe_get_evidence()`:
+
+1. Call OE SDK public API to initialize verifier plugins.
+2. Call the new API to verify the evidence. The format id is not required because it is contained in the evidence buffer.
+
+Scenario 1 is for users who provide their own evidence structures (like #2999, user hopes to verify a raw sgx quote).
+
+Scenario 2 is for users who full rely in OESDK ecosystem. They can use oe_generate_evidence to generate an oe_evidence structure and verify it by oe_verify_evidence.
```

Comment:
```suggestion
Scenario 2 is for users who use the existing OESDK everywhere. They can use oe_generate_evidence to generate an oe_evidence structure and verify it by oe_verify_evidence.
```
(Rationale: The OESDK "ecosystem" may include multiple formats, and in particular I hope it will include standards in the future, where the current format returned by oe_generate_evidence runs counter to emerging standards.  Hence softening wording in my suggestion.)

Reply:
Thank, I will update that.